34 running experiments in console (#120)
rungalileo/docs-official: concepts/experiments/running-experiments-in-console.mdx

This section will guide you through the process of running experiments in the Galileo Console.

## Experiment Walkthrough

Follow these steps to test and improve your AI projects using [Galileo's Console UI](https://app.galileo.ai).

### Step 1: Select Project & Open Playground

In the [Galileo Console](https://app.galileo.ai), use the **drop down menu in the top-left** to select the project you would like to experiment with. Or, create a new project.

Then, click the **"Open Playground"** button to access the Galileo Console.

![app.galileo.ai](/images/console-ui/home-navigation.png)

### Step 2: Select Model & Enter API Keys

In the Galileo Console, **select a model** using the "Model" drop down menu.

> Some models require that you **enter your corresponding API key**. Visit their respective API platforms to obtain your keys.

Reviewer: ```suggestion
Some models require that you **enter your corresponding API key**. Visit their respective API platforms to obtain your keys, then add it using the [integrations page](https://app.galileo.ai/settings/integrations) in the Galileo console.
```

Revert removing this please

* **Frequency Penalty:** Reduces the likelihood of the model repeating the same tokens by penalizing frequent ones.
* **Presence Penalty:** Discourages the model from mentioning tokens that have already appeared, promoting new content.

![Configure Model Settings](/images/console-ui/model-settings.png)

### Step 4: Configure Prompt Data

There are **two ways** to set the prompt data for your experiment:

* **Option 1**: Add prompts and variables through the console UI. Ideal for quick tests.
* **Option 2**: Use datasets from past experiments or create new ones. Ideal for real, fully-configured experiments.

### Step 4, Option 1: Add Prompt & Variables

In the Editor section, **add your prompt**.

In your prompt, you can use **variable names with curly brackets** (e.g. `{{variable_name}}`). Add new variable options with the **new tab icon** next to "Variable Set" and fill them in with different values to be used in place of your variable.

You can also use **nested variables** by entering JSON formatted key-value pairs into the "Variable Set" text input field. Then, refer to their values with either `{{key}}` or `{{input.key}}`

> ```

Reviewer: ```suggestion
```json
```

{
  "pepperoni": "pepperoni pizza",
  "anchovy": "pizza with anchovies"
}
```
In this example, both `{{pepperoni}}` and `{{input.pepperoni}}` will result in "pepperoni pizza" being used in the prompt. This approach is great for testing how changing individual words in a prompt structure affects outputs.

Add new messages beyond the initial prompt with the "+ Add Message" button **below the prompt field**. New messages can be from the user or the model ("system").

![Add Variables](/images/console-ui/variables-example.png)

### Step 4, Option 2: Add Dataset

Click the "Add Dataset" button to **choose a dataset** to be used by your model.

> The datasets listed are from your past experiments. You can also [add your own](/getting-started/experiments/datasets) by clicking "Create new dataset".

Reviewer: ```suggestion
The datasets listed are from your past experiments. You can also [add your own](/concepts/experiments/datasets) by clicking "Create new dataset".
```

> [Learn more about datasets →](/getting-started/experiments/datasets)

Reviewer: ```suggestion
[Learn more about datasets →](/concepts/experiments/datasets)
```

![Log Results](/images/console-ui/experiment-name.png)

### Step 10: Continue Experimenting!

That's it! Now, further customize and configure your experiment to meet your testing goals. Log your experiment results, and create new projects to try out different configurations.

If you encounter any errors, visit our [Common Errors guide](/references/faqs/errors).

## Experiment Settings & Options

![Continue Experimenting!](/images/console-ui/labeled-console-ui.png)

1. **Model Select** - choose model to be used with prompt/dataset (and enter API keys if necessary)
2. **Model Settings** - adjust model-specific settings. [Learn more about model settings →](#step-3-configure-model-settings)
3. **Message Originator** - select if the content of the prompt is from a user or from the model itself ("system").
4. **Prompt Entry** - add your prompt for the experiment. Use variables with curly brackets (e.g. `{{variable_name}}`) and add variable values in the variable entry field (#9). [Learn more about prompts & variables →](#step-4-option-1-add-prompt--variables)
5. **Add Message** - use a multi-prompt conversation in your experiment by adding new messages.

> 6. **Dataset Select** - instead of entering prompt(s), select a dataset of prompt data structures from a prior project, or add a new one. [Learn more about datasets →](/getting-started/experiments/datasets)

Reviewer: ```suggestion
6. **Dataset Select** - instead of entering prompt(s), select a dataset of prompt data structures from a prior project, or add a new one. [Learn more about datasets →](/concepts/experiments/datasets)
```
